src/expire.c

```diff
@@ -425,6 +425,11 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
      * Instead we take the other branch of the IF statement setting an expire
      * (possibly in the past) and wait for an explicit DEL from the master. */
     if (when <= mstime() && !server.loading && !server.masterhost) {
+        if (c->flags & CLIENT_LUA && !server.lua_replicate_commands) {
+            addReplyError(c,"Delete a key by *expire* is not allowed from scripts");
+            return;
+        }
```

Comment:
i'm not sure we can afford this.
it'll make the command non-deterministic (succeeding on the master at one time, and failing later).
@yossigo WDYT?